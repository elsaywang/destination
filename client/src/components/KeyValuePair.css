:local(.operator) {
    margin-right: 10px;
}

:local(.textField) {
    min-width: 250px;
}
