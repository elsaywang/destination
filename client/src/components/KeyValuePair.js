import React, { Component } from 'react';
import Autocomplete from '@react/react-spectrum/Autocomplete';
import Select from '@react/react-spectrum/Select';
import Textfield from '@react/react-spectrum/Textfield';
import Label from './common/Label';
import operatorOptions from '../constants/operatorOptions';
import styles from './KeyValuePair.css';
import classNames from 'classnames';
import { isValueValid } from '../utils/searchValidation';
import InlineErrorMessage from './common/InlineErrorMessage';

class KeyValuePair extends Component {
    constructor(props) {
        super(props);

        this.onSelectOperatorChange = this.props.onOperatorChange.bind(this, this.props.pair.id);
        this.onKeyChange = this.props.onKeyChange.bind(this, this.props.pair.id);
        this.onValueChange = this.props.onValueChange.bind(this, this.props.pair.id);
    }

    state = {
        autocompleteError: false,
        autocompleteErrorMessage: '',
    };

    getCompletions = key => {
        return fetch(`/api/signals/keys?search=${key}`)
            .then(response => {
                if (response.ok) {
                    this.setState({
                        autocompleteError: false,
                    });

                    return response.json();
                }

                throw new Error(response.statusText);
            })
            .catch(error => {
                this.setState({
                    autocompleteError: true,
                    autocompleteErrorMessage: error.message,
                });
            });
    };

    getKeysByReportSuiteId = () => {
        const { reportSuiteId } = this.props;

        return fetch(`/api/v1/report-suites/${reportSuiteId}/keys`)
            .then(response => {
                if (response.ok) {
                    this.setState({
                        autocompleteError: false,
                    });

                    return response.json();
                }

                throw new Error(response.statusText);
            })
            .then(suites => suites.keys.map(suite => suite.name))
            .catch(error => {
                this.setState({
                    autocompleteError: true,
                    autocompleteErrorMessage: error.message,
                });
            });
    };

    render() {
        const { id, key, value, operator } = this.props.pair;
        const { advanced } = this.props;
        const forKey = `key${id}`;
        const forValue = `value${id}`;
        const keyPlaceholder = `Enter a key${advanced ? ' or key name' : ''}`;
        const keyLabel = `Key${advanced ? ' or Key Name' : ''}`;
        const valuePlaceholder = 'Enter a value';
        const valueLabel = 'Value';

        return (
            <span data-test="key-value-pair">
                <Label value={keyLabel} labelFor={forKey} style={{ position: 'relative' }}>
                    <Autocomplete
                        className="key-search"
                        getCompletions={
                            advanced ? this.getKeysByReportSuiteId : this.getCompletions
                        }
                        value={key}
                        onChange={this.onKeyChange}>
                        <Textfield
                            className={styles.textField}
                            data-test="key-search-field"
                            id={forKey}
                            placeholder={keyPlaceholder}
                            invalid={this.state.autocompleteError}
                        />
                    </Autocomplete>
                    <InlineErrorMessage
                        isInvalid={this.state.autocompleteError}
                        errorMessage={this.state.autocompleteErrorMessage}
                    />
                </Label>
                <Select
                    className={classNames(styles.operator, 'operator')}
                    value={operator}
                    onChange={this.onSelectOperatorChange}
                    options={operatorOptions}
                />
                <Label value={valueLabel} labelFor={forValue} style={{ position: 'relative' }}>
                    <Textfield
                        className={classNames(styles.textField, 'value-search')}
                        data-test="value-search"
                        id={forValue}
                        value={value}
                        placeholder={valuePlaceholder}
                        onChange={this.onValueChange}
                        invalid={!isValueValid(this.props.pair)}
                    />
                    <InlineErrorMessage
                        isInvalid={!isValueValid(this.props.pair)}
                        errorMessage="It can only be numerical values when it's > or <."
                    />
                </Label>
            </span>
        );
    }
}

export default KeyValuePair;
