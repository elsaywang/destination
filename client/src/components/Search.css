:local(.row) {
    display: flex;
    align-items: center;
    margin-bottom: 1rem;
}

:local(.rowButtons) {
    margin-top: 25px;
}

:local(.ctas) {
    display: flex;
    align-items: end;
}

:local(.error) {
    float: right;
    margin-top: 8px;
}
