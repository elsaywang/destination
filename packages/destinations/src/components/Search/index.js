import SearchWrapper from './Search';

export default SearchWrapper;
