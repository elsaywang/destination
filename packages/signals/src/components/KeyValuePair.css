:local(.operator) {
    margin: 25px 10px 0 10px;
}

:local(.textField) {
    min-width: 300px;
}
