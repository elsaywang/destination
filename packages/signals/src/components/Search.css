:local(.row) {
    display: flex;
    align-items: center;
    margin-bottom: 1rem;
}

:local(.rowButtons) {
    margin-top: 25px;
    padding-left: 120px;
}

:local(.ctas) {
    display: flex;
    align-items: end;
}

:local(.error) {
    float: right;
    margin-top: 8px;
}

:local(.fieldLabel) {
    margin-right: 20px;
}

:local(.removeButton) {
    margin-left: 0 !important;
}

@media only screen and (min-width: 768px) {
    :local(.rowButtons) {
        padding-left: 225px;
    }
}
@media only screen and (min-width: 304px) and (min-width: 768px) {
    :local(.rowButtons) {
        padding-left: 290px;
    }
}
