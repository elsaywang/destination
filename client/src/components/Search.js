import React, { Component } from 'react';
import { GridColumn, GridRow } from '@react/react-spectrum/Grid';
import Well from '@react/react-spectrum/Well';
import Button from '@react/react-spectrum/Button';
import Select from '@react/react-spectrum/Select';
import Switch from '@react/react-spectrum/Switch';
import NumberInput from '@react/react-spectrum/NumberInput';
import AddCircle from '@react/react-spectrum/Icon/AddCircle';
import RemoveCircle from '@react/react-spectrum/Icon/RemoveCircle';
import AdvancedSearch from './AdvancedSearch';
import CustomDateRange from './CustomDateRange';
import KeyValuePair from './KeyValuePair';
import Label from './common/Label';
import InlineErrorMessage from './common/InlineErrorMessage';
import styles from './Search.css';
import statusOptions from '../constants/signalStatusOptions';
import { isFormValid } from '../utils/searchValidation';
import { getDateRangeOptionsWithinRetentionPolicy } from '../utils/dateRangeOptions';

class Search extends Component {
    getViewRecordsOptions = () =>
        getDateRangeOptionsWithinRetentionPolicy(this.props.maxSignalRetentionDays);

    renderKVPFields = pair => {
        const {
            keyValuePairs,
            advanced,
            source,
            onRemoveClick,
            onAddClick,
            onKeyChange,
            onOperatorChange,
            onValueChange,
        } = this.props;
        const validKeyValuePairsLimit = keyValuePairs.length < 3;
        const isLastPair = id => keyValuePairs[keyValuePairs.length - 1].id === id;
        const onRemove = () => onRemoveClick(pair.id);
        const reportSuiteId = source.reportSuiteIds;

        return (
            <GridRow key={pair.id}>
                <GridColumn size={12} className={styles.keyValuePairs}>
                    <KeyValuePair
                        key={pair.id}
                        pair={pair}
                        advanced={advanced}
                        reportSuiteId={
                            reportSuiteId && reportSuiteId.length ? reportSuiteId[0] : ''
                        }
                        onKeyChange={onKeyChange}
                        onOperatorChange={onOperatorChange}
                        onValueChange={onValueChange}
                    />
                    {isLastPair(pair.id) &&
                        validKeyValuePairsLimit && (
                            <Button
                                label="Add"
                                data-test="add-button"
                                onClick={onAddClick}
                                icon={<AddCircle />}
                                variant="action"
                                quiet
                            />
                        )}
                    {pair.id !== 0 && (
                        <Button
                            label="Remove"
                            data-test="remove-button"
                            onClick={onRemove}
                            icon={<RemoveCircle />}
                            variant="action"
                            quiet
                        />
                    )}
                </GridColumn>
            </GridRow>
        );
    };

    renderCustomDatepickers = () => {
        if (!this.props.isCustomDateRangeEnabled) {
            return null;
        }

        return (
            <CustomDateRange
                customStartDate={this.props.customStartDate}
                customEndDate={this.props.customEndDate}
                onCustomStartDateChange={this.props.onCustomStartDateChange}
                onCustomEndDateChange={this.props.onCustomEndDateChange}
                minCustomStartDateDaysAgo={this.props.maxSignalRetentionDays}
            />
        );
    };

    renderCTAs = () => (
        <GridColumn className={styles.ctas}>
            <Button
                label="Clear All"
                data-test="clear-all-button"
                onClick={this.props.onClearAll}
                quiet
                variant="secondary"
            />
            <Button
                label="Search"
                data-test="search-button"
                onClick={this.props.onSearch}
                variant="cta"
                disabled={!isFormValid(this.props)}
            />
        </GridColumn>
    );

    render() {
        return (
            <GridRow>
                <GridColumn size={12}>
                    <Switch
                        className="advanced-search-toggle"
                        onChange={this.props.onAdvancedSearchChange}
                        checked={this.props.advanced}
                        data-test="advanced-search-toggle"
                        aria-label="Advanced Search"
                        label="Advanced search for Adobe Analytics"
                    />
                    <div>
                        <InlineErrorMessage
                            isInvalid={this.props.errors.reportSuites.hasError}
                            errorMessage={this.props.errors.reportSuites.errorMessage}
                        />
                    </div>
                    <Well>
                        <div data-test="search-form">
                            {this.props.advanced && (
                                <AdvancedSearch
                                    enabled={this.props.advanced}
                                    onReportSuiteChange={this.props.onReportSuiteChange}
                                    onReportSuiteSelect={value =>
                                        this.props.onReportSuiteSelect(value)
                                    }
                                    sourceName={this.props.source.name}
                                    reportSuites={this.props.reportSuites}
                                />
                            )}

                            <GridRow>
                                <GridColumn size={12}>
                                    {this.props.keyValuePairs.map(this.renderKVPFields)}

                                    <GridRow>
                                        <GridColumn size={10}>
                                            <Label value="Signal Status">
                                                <Select
                                                    className="signal-status"
                                                    data-test="signal-status"
                                                    value={this.props.signalStatus}
                                                    onChange={this.props.onSignalStatusChange}
                                                    options={statusOptions}
                                                    quiet
                                                />
                                            </Label>

                                            <Label value="View Records For">
                                                <Select
                                                    className="view-records"
                                                    data-test="view-records"
                                                    value={this.props.viewRecordsFor}
                                                    onChange={this.props.onViewRecordsChange}
                                                    options={this.getViewRecordsOptions()}
                                                    quiet
                                                />
                                            </Label>

                                            {this.renderCustomDatepickers()}

                                            <Label value="Minimum Counts">
                                                <NumberInput
                                                    className="min-counts"
                                                    data-test="min-counts"
                                                    onChange={this.props.onMinEventFiresChange}
                                                    value={this.props.minEventFires}
                                                    min={this.props.eventFiresMinimum}
                                                    step={this.props.eventFiresStep}
                                                    quiet
                                                />
                                            </Label>
                                        </GridColumn>

                                        {this.renderCTAs()}
                                    </GridRow>
                                    <InlineErrorMessage
                                        className={styles.error}
                                        isInvalid={this.props.errors.searchForm.hasError}
                                        errorMessage={this.props.errors.searchForm.errorMessage}
                                    />
                                </GridColumn>
                            </GridRow>
                        </div>
                    </Well>
                </GridColumn>
            </GridRow>
        );
    }
}

export default Search;
