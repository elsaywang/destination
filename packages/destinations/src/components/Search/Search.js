import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { connect } from 'react-redux';
import Search from '@react/react-spectrum/Search';
import { fetchDestinations, applySearch } from '../../redux/actions/destinations';
import styles from '../../components/Layout';

class SearchWrapper extends Component {
    fetchDestinationsDebounced = _.debounce(this.props.fetchDestinations, 500);

    render() {
        const { destinations, applySearch } = this.props;
        return (
            <Search
                className={styles.search}
                id={`destinations-search`}
                placeholder="Search"
                onChange={query => {
                    applySearch(query);
                    this.fetchDestinationsDebounced();
                }}
                value={destinations.searchFormText}
            />
        );
    }
}

SearchWrapper.propTypes = {
    fetchDestinations: PropTypes.func,
};

export default connect(
    destinations => destinations,
    { fetchDestinations, applySearch },
)(SearchWrapper);
