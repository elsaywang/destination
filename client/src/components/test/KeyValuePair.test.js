import React from 'react';
import { shallow } from 'enzyme';
import operators from '../../constants/operatorOptions';
import KeyValuePair from '../KeyValuePair';
import Autocomplete from '@react/react-spectrum/Autocomplete';
import Select from '@react/react-spectrum/Select';
import Search from '@react/react-spectrum/Search';
import Textfield from '@react/react-spectrum/Textfield';

describe('<KeyValuePair /> component', () => {
    const mockFn = jest.fn();
    const pair = {
        id: 0,
        key: '',
        operator: '=',
        value: '',
    };
    const wrapper = shallow(
        <KeyValuePair
            pair={pair}
            onKeySelect={mockFn}
            onOperatorChange={mockFn}
            onValueChange={mockFn}
        />,
    );

    describe('rendering', () => {
        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <Autocomplete /> and <Textfield /> child for key / key name search', () => {
            const autocomplete = wrapper.find(Autocomplete);

            expect(autocomplete.filter('.key-search').exists()).toBe(true);
            expect(autocomplete.find(Textfield).exists()).toBe(true);
        });

        it('renders <Select /> with operator options', () => {
            const operatorSelect = wrapper.find(Select).filter('.operator');
            const operatorOptions = operatorSelect.props().options;

            expect(operatorSelect.exists()).toBe(true);
            expect(operatorOptions).toMatchObject(operators);
        });

        it('renders <Search /> for value search', () => {
            expect(
                wrapper
                    .find(Search)
                    .filter('.value-search')
                    .exists(),
            ).toBe(true);
        });
    });
});
