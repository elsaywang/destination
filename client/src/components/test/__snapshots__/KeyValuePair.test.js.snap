// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KeyValuePair /> component rendering matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <KeyValuePair
    onKeySelect={[MockFunction]}
    onOperatorChange={[MockFunction]}
    onValueChange={[MockFunction]}
    pair={
      Object {
        "id": 0,
        "key": "",
        "operator": "==",
        "value": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Label
          labelFor="key0"
          value="Key"
        >
          <Autocomplete
            allowCreate={false}
            className="key-search"
            getCompletions={[Function]}
            onSelect={[Function]}
            value=""
          >
            <Textfield
              data-test="key-search-field"
              disabled={false}
              id="key0"
              invalid={false}
              placeholder="Type a key"
              readOnly={false}
              required={false}
              value=""
            />
          </Autocomplete>
        </Label>,
        <Select
          className="operator"
          onChange={[Function]}
          options={
            Array [
              Object {
                "label": "==",
                "value": "==",
              },
              Object {
                "label": ">",
                "value": ">",
              },
              Object {
                "label": "<",
                "value": "<",
              },
              Object {
                "label": ">=",
                "value": ">=",
              },
              Object {
                "label": "<=",
                "value": "<=",
              },
              Object {
                "label": "contains",
                "value": "contains",
              },
            ]
          }
          value="=="
        />,
        <Label
          labelFor="value0"
          value="Value"
        >
          <Textfield
            className="value-search"
            data-test="value-search"
            disabled={false}
            id="value0"
            invalid={false}
            onChange={[Function]}
            placeholder="Type a value"
            readOnly={false}
            required={false}
            value=""
          />
        </Label>,
      ],
      "data-test": "key-value-pair",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Autocomplete
            allowCreate={false}
            className="key-search"
            getCompletions={[Function]}
            onSelect={[Function]}
            value=""
          >
            <Textfield
              data-test="key-search-field"
              disabled={false}
              id="key0"
              invalid={false}
              placeholder="Type a key"
              readOnly={false}
              required={false}
              value=""
            />
          </Autocomplete>,
          "labelFor": "key0",
          "value": "Key",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowCreate": false,
            "children": <Textfield
              data-test="key-search-field"
              disabled={false}
              id="key0"
              invalid={false}
              placeholder="Type a key"
              readOnly={false}
              required={false}
              value=""
            />,
            "className": "key-search",
            "getCompletions": [Function],
            "onSelect": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data-test": "key-search-field",
              "disabled": false,
              "id": "key0",
              "invalid": false,
              "placeholder": "Type a key",
              "readOnly": false,
              "required": false,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "operator",
          "onChange": [Function],
          "options": Array [
            Object {
              "label": "==",
              "value": "==",
            },
            Object {
              "label": ">",
              "value": ">",
            },
            Object {
              "label": "<",
              "value": "<",
            },
            Object {
              "label": ">=",
              "value": ">=",
            },
            Object {
              "label": "<=",
              "value": "<=",
            },
            Object {
              "label": "contains",
              "value": "contains",
            },
          ],
          "value": "==",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Textfield
            className="value-search"
            data-test="value-search"
            disabled={false}
            id="value0"
            invalid={false}
            onChange={[Function]}
            placeholder="Type a value"
            readOnly={false}
            required={false}
            value=""
          />,
          "labelFor": "value0",
          "value": "Value",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "value-search",
            "data-test": "value-search",
            "disabled": false,
            "id": "value0",
            "invalid": false,
            "onChange": [Function],
            "placeholder": "Type a value",
            "readOnly": false,
            "required": false,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Label
            labelFor="key0"
            value="Key"
          >
            <Autocomplete
              allowCreate={false}
              className="key-search"
              getCompletions={[Function]}
              onSelect={[Function]}
              value=""
            >
              <Textfield
                data-test="key-search-field"
                disabled={false}
                id="key0"
                invalid={false}
                placeholder="Type a key"
                readOnly={false}
                required={false}
                value=""
              />
            </Autocomplete>
          </Label>,
          <Select
            className="operator"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "label": "==",
                  "value": "==",
                },
                Object {
                  "label": ">",
                  "value": ">",
                },
                Object {
                  "label": "<",
                  "value": "<",
                },
                Object {
                  "label": ">=",
                  "value": ">=",
                },
                Object {
                  "label": "<=",
                  "value": "<=",
                },
                Object {
                  "label": "contains",
                  "value": "contains",
                },
              ]
            }
            value="=="
          />,
          <Label
            labelFor="value0"
            value="Value"
          >
            <Textfield
              className="value-search"
              data-test="value-search"
              disabled={false}
              id="value0"
              invalid={false}
              onChange={[Function]}
              placeholder="Type a value"
              readOnly={false}
              required={false}
              value=""
            />
          </Label>,
        ],
        "data-test": "key-value-pair",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Autocomplete
              allowCreate={false}
              className="key-search"
              getCompletions={[Function]}
              onSelect={[Function]}
              value=""
            >
              <Textfield
                data-test="key-search-field"
                disabled={false}
                id="key0"
                invalid={false}
                placeholder="Type a key"
                readOnly={false}
                required={false}
                value=""
              />
            </Autocomplete>,
            "labelFor": "key0",
            "value": "Key",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowCreate": false,
              "children": <Textfield
                data-test="key-search-field"
                disabled={false}
                id="key0"
                invalid={false}
                placeholder="Type a key"
                readOnly={false}
                required={false}
                value=""
              />,
              "className": "key-search",
              "getCompletions": [Function],
              "onSelect": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "data-test": "key-search-field",
                "disabled": false,
                "id": "key0",
                "invalid": false,
                "placeholder": "Type a key",
                "readOnly": false,
                "required": false,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "operator",
            "onChange": [Function],
            "options": Array [
              Object {
                "label": "==",
                "value": "==",
              },
              Object {
                "label": ">",
                "value": ">",
              },
              Object {
                "label": "<",
                "value": "<",
              },
              Object {
                "label": ">=",
                "value": ">=",
              },
              Object {
                "label": "<=",
                "value": "<=",
              },
              Object {
                "label": "contains",
                "value": "contains",
              },
            ],
            "value": "==",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Textfield
              className="value-search"
              data-test="value-search"
              disabled={false}
              id="value0"
              invalid={false}
              onChange={[Function]}
              placeholder="Type a value"
              readOnly={false}
              required={false}
              value=""
            />,
            "labelFor": "value0",
            "value": "Value",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "value-search",
              "data-test": "value-search",
              "disabled": false,
              "id": "value0",
              "invalid": false,
              "onChange": [Function],
              "placeholder": "Type a value",
              "readOnly": false,
              "required": false,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
